app/Repositories/PostRepository.php

```diff
@@ -87,18 +87,30 @@ public function create(array $data, $signupId, $authenticatedUserRole = null)
             'action' => isset($data['action']) ? $data['action'] : null,
             'url' => $fileUrl,
             'text' => isset($data['text']) ? $data['text'] : null,
-            'status' => 'pending',
             'source' => token()->client(),
             'source_details' => isset($data['source_details']) ? $data['source_details'] : null,
             'details' => isset($data['details']) ? $data['details'] : null,
         ]);
 
+        // If this is a share-social type post, auto-accept.
+        $post->status = $post->type === 'share-social' ? 'accepted' : 'pending';
+
         $isAdmin = auth()->user() && auth()->user()->role === 'admin';
         $hasAdminScope = in_array('admin', token()->scopes());
 
         // Admin users may provide a source, status, and created_at when uploading a post.
         if ($isAdmin || $hasAdminScope) {
-            $post->status = isset($data['status']) ? $data['status'] : 'pending';
+            // If the admin sets a custom status, set this status.
+            if (isset($data['status'])) {
+                $post->status = $data['status'];
+            // If the admin doesn't set a custom status and it is a share-social post, auto-accept it.
+            } elseif (! isset($data['status']) && $post->type === 'share-social') {
+                $post->status = 'accepted';
+            // Otherwise, the status should be pending.
+            } else {
+                $post->status = 'pending';
+            }
```

Comment:
I'm a little confused here. Doesn't the `share-social` situation get taken care of above? And don't we want the status to be `pending` even if `$isAdmin || $hasAdminScope` is `false`?

Reply:
>Doesn't the share-social situation get taken care of above? 

This block deals with the following logic: (I'm not sure how often this will happen since admins probably can't upload `share-social` posts for users) if an admin sets a custom status for a `share-social` post, accept that status instead of auto-accepting like we would normally do. If we didn't update this, the old logic would change the `status` back to `pending` after auto-accepting if the admin doesn't set a custom status.

>And don't we want the status to be pending even if $isAdmin || $hasAdminScope is false

We want to auto-accept `share-social` posts when they are created!

@katiecrane  lmk if that still doesn't make sense or if you have more q's!